### DISCUSSION TITLE
Rollout strategy

### DISCUSSION THREAD

**Discussion by dvic** [basecamp/kamal#1209] [Q&A] [answered]
Is there a way to use a 'recreate' rollout strategy? I have some deployments that acquire a lock on the database and therefore I can't have two or more instances running at the same time.

**Comment by tuladhar** [comment]
You may want to checkout the [boot configuration](https://kamal-deploy.org/docs/configuration/booting/) which provides options for controlling the deployment strategy.

**Reply by dvic** [reply]
This only controls the deployment strategy across nodes, right? In my case I need to control the deployment strategy on a single node.

**Reply by tuladhar** [reply, marked as answer]
Since Kamal automatically boots a new container for zero-downtime deployments, a possible workaround for your situation would be to run `kamal app stop` followed by `kamal deploy`. This process effectively recreates the application.

Alternatively, you could utilize the `pre-deploy` hook as a workaround to run `kamal app stop`.

**Reply by dvic** [reply]
> Since Kamal automatically boots a new container for zero-downtime deployments, a possible workaround for your situation would be to run `kamal app stop` followed by `kamal deploy`. This process effectively recreates the application.

Yes, that was what I was doing currently.

> Alternatively, you could utilize the `pre-deploy` hook as a workaround to run `kamal app stop`.

Ah of course, does this hook run on the server or locally from the machine where kamal deploy was run from?

Thanks for the help!
